api/api_ethereum.go

```diff
@@ -679,3 +729,36 @@ func (api *EthereumAPI) Accounts() []common.Address {
 	// TODO-Klaytn: Not implemented yet.
 	return nil
 }
+
+// rpcMarshalHeader marshal block header as Ethereum compatible format
+func (api *EthereumAPI) rpcMarshalHeader(head *types.Header) map[string]interface{} {
+	proposer, err := api.publicKlayAPI.b.Engine().Author(head)
+	if err != nil {
+		logger.Error("Failed to fetch author during marshaling header: %s", err.Error(), nil)
```

Comment:
Parameters are not proper for the logger

Reply:
Thanks for your feedback.

Applied your feedback. Now log is printed well.
```
ERROR[01/10,10:32:26 +09] [3] Failed to fetch author during marshaling header  err="invalid istanbul header extra-data"
```

Reply:
It returns an empty proposer when err is not nil. Is it okay?

Reply:
@aidan-kwon 
As you said, an empty proposer will be served if error happens at `Author()`.
Although it is not perfect information(miner field would be empty address `0x0000000000000000000000000000000000000000` in error case), I thought we should provide information about other fields if possible.

Reply:
@aidan-kwon 
So, there could be 2 options about this issue.

**Option 1**: Provide header even if a miner field is empty address
In this case, user will be received header like below.
```json
{
  baseFeePerGas: "0x0",
  difficulty: "0x1",
  extraData: "0x",
  gasLimit: "0x3b9ac9ff",
  gasUsed: "0x0",
  hash: "0xc8036293065bacdfce87debec0094a71dbbe40345b078d21dcc47adb4513f348",
  logsBloom: "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  miner: "0x0000000000000000000000000000000000000000",
  mixHash: "0x0000000000000000000000000000000000000000000000000000000000000000",
  nonce: "0x0000000000000000",
  number: "0x3",
  parentHash: "0x8c73a925cf0ec8019162b32bb75b6f1df706213df4468de131dfde9612e33331",
  receiptsRoot: "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470",
  sha3Uncles: "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
  size: "0x2cc",
  stateRoot: "0xae2f08edc892858210ce258f3bd0d70fde29eefe85e16e3535802a36e15a974c",
  timestamp: "0x61d53853",
  totalDifficulty: "0x4",
  transactionsRoot: "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
}
```

**Option2**: Do not provide header
In this case, user got a `null` about the header because fetching apis in Ethereum namespace does not return error, so user will always be got `null` as result.

Reply:
I think Option2 is better. All block signatures should be recoverable, and `miner` is the field Klaytn should provide the correct value. It's not the field dummy value is allowed. 
Even though Ethereum doesn't return errors for this API, I think returning errors will not make formatting errors because SDK usually defines a formatter for an entire API instead of each API. We can build rules like the following. 
- For the error cases Ethereum returns nil, Klaytn returns nil also. 
- For the error cases Ethereum doesn't expect, Klaytn returns errors.

Reply:
@aidan-kwon 

Great. I had a few questions about your comment.
There are two apis returning Block Header in this PR.
```go
func (api *EthereumAPI) GetHeaderByNumber(ctx context.Context, number rpc.BlockNumber) (map[string]interface{}, error) {}

func (api *EthereumAPI) GetHeaderByHash(ctx context.Context, hash common.Hash) map[string]interface{} {}
```

I understood your comment like below.
* `GetHeaderByNumber` should return error with `nil` at block header when error happened during fetching miner address.
* `GetHeaderByHash` should return `nil` because it only returns header, not err by it's function signature.

Am I understanding correctly?

Reply:
@aidan-kwon 
Applied your feedback [here](https://github.com/klaytn/klaytn/commit/84e603725f56d282ee29a8a48538a2cdb8dce47b).

Now if error happened during fetching miner address,
user would will got response like below.

```shell
$ curl http://localhost:8551 \
-X POST \
-H "Content-Type: application/json" \
-d '{"jsonrpc":"2.0","method":"eth_getHeaderByNumber","params":["0x4"],"id":1}'
{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"recovery failed"}}
```
> It returns error because the function signature of `eth_getHeaderByNumber` returns both header and error.


```shell
$ curl http://localhost:8551 \
-X POST \
-H "Content-Type: application/json" \
-d '{"jsonrpc":"2.0","method":"eth_getHeaderByHash","params":["0xcb2a163d2658d718e031c513088ef2064a2deea8320cff4a3ed1d481cfd1cbe9"],"id":1}'
{"jsonrpc":"2.0","id":1,"result":null}
```
> It returns null as result because the function signature of `eth_getHeaderByHash` returns only header.